### PR TITLE
adds documentation to insight and adds it to chainslist

### PR DESCRIPTION
**Pull request by iuwqyir**
## Problem solved

[BLOCK-325
](https://linear.app/thirdweb/issue/BLOCK-325/add-insight-to-thirdweb-chainslist)

Short description of the bug fixed or feature added

---

## PR-Codex overview
This PR introduces a new feature called `Insight`, which enhances the application by providing a dedicated section for querying and analyzing blockchain data. It includes new components, routes, and icons related to this feature.

### Detailed summary
- Added `Insight` link in the header and sidebar.
- Created `InsightSection` in `page.tsx`.
- Developed `sidebar` and `layout` for `Insight`.
- Introduced `InfraInsightIcon` and `InsightIcon`.
- Updated metadata for `Insight` across various pages.
- Documented use cases and getting started guides for `Insight`.
- Added `Insight` SVG icon and corresponding components.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by iuwqyir** [comment]
* **#5376** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/5376?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @iuwqyir and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/dashboard/src/app/(dashboard)/(chain)/components/server/products.ts [hunk 1]
```diff
@@ -49,6 +50,13 @@ export const products = [
     description: "Point of sale solution for bridging, onramping & swapping",
     link: "https://portal.thirdweb.com/connect/pay/overview",
   },
+  {
+    name: "Insight",
+    id: "insight",
+    icon: InsightIcon,
+    description: "Query, transform and analyze blockchain data",
+    link: "https://portal.thirdweb.com/insight",
+  },
```
Comment (MananTank) [inline apps/dashboard/src/app/(dashboard)/(chain)/components/server/products.ts:59]: In the chainlist page in dashboard, No chains are shown to be supporting insight service - is that correct?

File: apps/portal/src/app/insight/layout.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    <DocLayout sideBar={sidebar} editPageButton={true}>
+      {props.children}
+    </DocLayout>
+  );
+}
+
+export const metadata = createMetadata({
+  title: "Insight",
+  description: "Query, transform and analyze blockchain data",
+});
```
Comment (MananTank) [inline apps/portal/src/app/insight/layout.tsx:16]: Please also set `image` key for adding an OG image

Example: 
```
image: {
 title: "Insight",
 icon: "thirdweb"
}
```

You can also add a new icon for "insight" in `createMetadata` function

Comment (iuwqyir) [reply apps/portal/src/app/insight/layout.tsx:20]: Added icon for insight and updated the image prop in metadata

File: apps/dashboard/src/app/(dashboard)/(chain)/components/server/products.ts (hunk 1 above)
Comment (iuwqyir) [reply apps/dashboard/src/app/(dashboard)/(chain)/components/server/products.ts:59]: It's querying dev api, which hasn't indexed any chains. If I run it locally against prod api, I do see chains with insight enabled

**Comment by MananTank** [comment]
Deferring the dashboard review to @jnsdls here

**Review by MananTank — APPROVED** [review]

**Pull request merged**